### PR TITLE
Refactor quattor/schema

### PR DESCRIPTION
**Pull request by stdweird**
`quattor/schema` can't be used from the `template-library-core` repository due to the missing `monitoring/lemon/schema`. 
With the refactor,it is now possible for components to include only the component relevant types and nothing else, so we can unittest the templates shipped by e.g. the components with only dependency on this repo.

### CODE REVIEW

**Comment by stdweird** [comment]
If https://github.com/quattor/maven-tools/pull/42 gets merged/released, this PR will also be required, because component schemas can't be validated without it.
@jouvin any further remarks/objections?

**Comment by jrha** [comment]
Can you update the annotations to use `desc = blah`, then I think this is good to go.

File: quattor/blockdevices.pan [hunk 1]
```diff
@@ -1,41 +1,27 @@
...
-# type definition for block devices.
-#
-#
-#
-# DESCRIPTION: @DESCRIPTION@
-
-#
-############################################################
+@{
+    Type definitions for block devices.
```
Comment (jrha) [inline quattor/blockdevices.pan]:
```pan
desc = Type definitions for block devices
```

File: quattor/blockdevices.pan [hunk 2]
```diff
@@ -61,19 +46,19 @@ type blockdevices_partition_type = {
 	"flags" ? blockdevices_partition_flags
 };
 
-#
-# Software RAID using the MD device
-#
+@{
+    Software RAID using the MD device
```
Comment (jrha) [inline quattor/blockdevices.pan]: `desc = Software RAID using the MD device`

etc.

File: quattor/blockdevices.pan (hunk 1 above)
Comment (stdweird) [inline quattor/blockdevices.pan]: reading http://quattor-pan.readthedocs.org/en/latest/pan-book/pan-book.html?highlight=annotation, `desc` seems to be used for functions (and require the `@documentation{}`).

**Comment by stdweird** [comment]
@jrha i updated according to panbook.

**Comment by jrha** [comment]
Nice, looks good now, @jouvin please merge if you are happy.

**Pull request merged**